src/components/Api.js

```diff
@@ -0,0 +1,82 @@
+import Card from "./Card.js";
+import UserInfo from "./UserInfo.js";
+
+export default class Api {
+  constructor({ baseUrl, headers }) {
+    this._baseUrl = baseUrl;
+    this._headers = headers;
+  }
+
+  _handleResponse(res) {
+    if (res.ok) {
+      return res.json();
+    } else {
+      return Promise.reject(`Error: ${res.status}`);
+    }
+  }
+
+  getUserInfo() {
+    return fetch(`${this._baseUrl}/users/me`, {
+      headers: this._headers,
+    }).then(this._handleResponse);
+  }
+
+  getInitialCards() {
+    return fetch(`${this._baseUrl}/cards`, {
+      headers: this._headers,
+    }).then(this._handleResponse);
+  }
+
+  editProfile() {
+    return fetch(`${this._baseUrl}/users/me`, {
+      method: "PATCH",
+      headers: this._headers,
+      body: JSON.stringify({
+        name: "Marie Skłodowska Curie",
+        about: "Physicist and Chemist",
+      }),
+    });
```

Comment:
Needs correction: Here you should also handle the response and use the response in index.js in `then` block